inference-engine/src/mkldnn_plugin/emitters/jit_snippets_emitters.hpp

```diff
@@ -0,0 +1,641 @@
+    std::vector<int64_t> output_dims = {};
+};
+///
+/// \brief    Kernel is the only entry point to Codogen Jit compilation. Kernel calculates appropriate data offsets,
+/// and invokes enclosed outer Tiles. Only 2d Tiles are currently supported, so the emitters should
+/// be organized in the following way:
+/// KernelEmitter {          /* entry point */
+///     TileEmitter {        /* outer tile */
+///         TileEmitter {    /* inner vector tile */
+///             ...          /* All the necessary Load/Strore/elementwise emitters */
+///         }
+///         TileEmitter {    /* inner scalar tile for tail processing */
+///             ...          /* All the necessary Load/Strore/elementwise emitters */
+///         }
+///     }
+/// }
+/// Note that Kernel params are passed directly to the emit_code(). The vector of inputs should contain 2 arguments, the
+/// output vector should be empty. Input parameters
+///
+/// \param      num_inputs       The number of the node inputs
+/// \param      num_outputs      The number of the node outputs
+///
+// Todo: Scheduler dims and offsets are currently calculated in MKLDNN Subgraph node and passed to the KernelEmitter.
+//  However, it seems more natural to calculate all the offsets right in the Kernel op, because the calculation is
+//  not device-specific. It is based only on input/output dims (which we already know) and harness num dims
+//  (which we should pass from the plugin). It seems also better to wrap the enclosed emitters in tiles in the Kernel op
+//  and avoid creating empty tiles.
+class KernelEmitter : public jit_emitter {
+public:
+    KernelEmitter(mkldnn::impl::cpu::x64::jit_generator* h, mkldnn::impl::cpu::x64::cpu_isa_t isa,
+    const std::shared_ptr<ov::Node>& n)
+    : jit_emitter(h, isa, n) {
+        const auto kernel = ov::as_type_ptr<ngraph::snippets::op::Kernel>(n);
+        if (!kernel)
+            IE_THROW() << "KernelEmitter invoked with invalid op argument";
+        if (!kernel->compile_params)
+            IE_THROW() << "KernelEmitter invoked without compile_params";
+        code = kernel->region;
+        jcp = *reinterpret_cast<const jit_snippets_compile_args*>(kernel->compile_params);
+    }
+
+    size_t get_inputs_num() const override {return 0;}
+
+    void emit_code(const std::vector<size_t> &in, const std::vector<size_t> &out,
+              const std::vector<size_t> &pool = {}, const std::vector<size_t> &gpr = {}) const override {
+        validate_arguments(in, out, pool, gpr);
+        emit_impl(in, out, pool, gpr, nullptr);
+    }
+    void validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out,
+                            const std::vector<size_t> &pool = {}, const std::vector<size_t> &gpr = {}) const override {
+        if (in.size() != 2)
+            IE_THROW() << "Codegen Kernel got invalid number of inputs. Expected 2, got " << in.size();
```

Comment:
"Codegen Kernel" -> "KernelEmitter"

Reply:
Corrected, also for TileEmitter and ScalarEmitter.